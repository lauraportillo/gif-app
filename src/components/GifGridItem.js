import React from 'react'
import PropTypes from 'prop-types'
import '../stylesheets/GifGridItem.scss';

const GifGridItem = ({ title, url }) => {

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img className='card__image' src={url} alt={title} />
            <p className='card__text'>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


export default GifGridItem;