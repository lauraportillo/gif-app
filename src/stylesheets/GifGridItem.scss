.card{
    align-content: center;
    border:1px solid grey;
    border-radius: 6px;
    margin-bottom: 10px;
    margin-right: 10px;
    overflow: hidden;
}

.card__image{
    max-height: 170px;

}

.card__text{
    padding: 5px;
    text-align: center;
}