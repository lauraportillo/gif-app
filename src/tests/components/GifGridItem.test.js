import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGridItem from '../../components/GifGridItem';

describe('Testing <GifGridItem />', () => {

    test('should show the component correctly', () => {

        const wrapper = shallow(<GifGridItem />);

        expect(wrapper).toMatchSnapshot();

    });

});