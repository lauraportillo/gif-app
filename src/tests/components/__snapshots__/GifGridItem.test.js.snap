// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    className="card__image"
  />
  <p
    className="card__text"
  />
</div>
`;
